File: datacode_interview_task.py
# Python program to read in json files for
# language conversion application

import json

# open the english, french and pirate json files
with open("languages-english.json") as f1, open("languages-french.json") as f2, open("languages-pirate.json") as f3:
    # create dicts from json objs
    data1 = json.load(f1)
    data2 = json.load(f2)
    data3 = json.load(f3)

# function takes a language key string as an arg
def string_translator(keyString):
    # find out which file we are going to use
    print('English or French or Pirate?')
    selection = input().upper()
    # access the english file
    if selection == 'ENGLISH':
        for i in data1[keyString.upper()]:
            print(i, end='')
    # access the french file
    elif selection == 'FRENCH':
        for i in data2[keyString.upper()]:
            print(i, end='')
    # access the pirate file
    elif selection == 'PIRATE':
        try:
            for i in data3[keyString.upper()]:
                print(i, end='')
        # if the keyString isn't found, use english string
        except KeyError:
            for i in data1[keyString.upper()]:
                print(i, end='')  
    # handle invalid inputs
    else:
        print('Sorry, translation not available')

# close the files
f1.close()
f2.close()
f3.close()

File: datacode_interview_task.php
<?php 

// open files for reading and parse in the whole json file
// save the json files to the given variable $fileX
$file1 = fopen("languages-english.json","r");
$file2 = fopen("languages-french.json", "r");
$file3 = fopen("languages-pirate.json", "r");
fread($file1, filesize("languages-english.json"));
fread($file2, filesize("languages-french.json"));
fread($file3, filesize("Languages-pirate.json"));

// add the json file to an array
$array1 = json_decode($file1, true);
$file1 = json_decode($json);

$array2 = json_decode($file2, true);
$file2 = json_decode($json);

$array3 = json_decode($file3, true);
$file3 = json_decode($json);

// select which language to translate to (change as needed)
$selection = "English";

// based on the selection given, search the correct array for the translation
function stringTranslator($keyString){
	if $selection == 'English';{
		print $key = array_search ($keyString, $array1);	
	} elseif $selection == 'French';{
		print $key = array_search ($keyString, $array2);
	} elseif $selection == 'Pirate';{
		// if the keyString can be found in the pirate array do the translation
		try{
			print $key = array_search ($keyString, $array3);
		// if the keyString cannot be found in the array, return the English translation
		} catch (OutOfBoundsException $exception){
			print $key = array_search ($keyString, $array1);
			}
	// if the language is not recognised, show this error message
	} else {
		print "Translation not found"
	}
}

// close all the files
fclose($file1)
fclose($file2)
fclose($file3)

?>
